evalml/pipelines/time_series_classifcation_pipelines.py

```diff
@@ -0,0 +1,204 @@
+            y (ww.DataColumn, pd.Series, np.ndarray, None): The target training targets of length [n_samples]
+            objective (Object or string): The objective to use to make predictions
+
+        Returns:
+            pd.Series: Predicted values.
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        n_features = max(len(y), X.shape[0])
+        predictions = self._predict(X, y, objective=objective, pad=False)
+        predictions = pd.Series(self._decode_targets(predictions))
+        return pad_with_nans(predictions, max(0, n_features - predictions.shape[0]))
+
+    def predict_proba(self, X, y=None):
+        """Make probability estimates for labels.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): Data of shape [n_samples, n_features]
+
+        Returns:
+            pd.DataFrame: Probability estimates
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        y_encoded = self._encode_targets(y)
+        features = self.compute_estimator_features(X, y_encoded)
+        proba = self.estimator.predict_proba(features.dropna(axis=0, how="any"))
+        proba.columns = self._encoder.classes_
+        return pad_with_nans(proba, max(0, features.shape[0] - proba.shape[0]))
+
+    def _compute_predictions(self, X, y, objectives):
+        """Compute predictions/probabilities based on objectives."""
+        y_predicted = None
+        y_predicted_proba = None
+        if any(o.score_needs_proba for o in objectives):
+            y_predicted_proba = self.predict_proba(X, y)
+        if any(not o.score_needs_proba for o in objectives):
+            y_predicted = self._predict(X, y, pad=True)
+        return y_predicted, y_predicted_proba
+
+    def score(self, X, y, objectives):
+        """Evaluate model performance on current and additional objectives.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): Data of shape [n_samples, n_features]
+            y (pd.Series, ww.DataColumn): True labels of length [n_samples]
+            objectives (list): Non-empty list of objectives to score on
+
+        Returns:
+            dict: Ordered dictionary of objective scores
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        objectives = [get_objective(o, return_instance=True) for o in objectives]
+
+        y_encoded = self._encode_targets(y)
+        y_shifted = y_encoded.shift(-self.gap)
+        y_pred, y_pred_proba = self._compute_predictions(X, y, objectives)
+        non_nan_mask = _get_rows_without_nans(y_shifted, y_pred, y_pred_proba)
+        if y_pred is not None:
```

Comment:
This is a bit ugly but depending on the objectives that are passed in, `y_pred` or `y_pred_proba` can be `None` so we can only drop nans if possible. 

I've thought about avoiding this complexity by having `_compute_predictions` return un-padded predictions so that we don't have to worry about dropping anything. However, that doesn't really work because the predictions and `y_shifted` will have different number of nans depending on the values of `gap` and the number of delayed features computed in the pipeline. The least complex way of knowing what the common non-nan subset is by looking at the nans in the predictions and targets.

In short, we may be able to hide this complexity in a helper function but I don't think we can avoid it.